add node log page (#443)
klaytn/klaytn-docs: docs/node/node-log.md

# Node Log <a id="node-log"></a>

> This page details some important logs of the Klaytn node logs.

Reviewer: ```suggestion
This page details some important or frequently asked logs of the Klaytn node logs.
```

If the Klaytn log is modified or newly added/deleted, please edit this page as well.

For more detailed information about log type, you can refer [log_modules.go](https://github.com/klaytn/klaytn/blob/dev/log/log_modules.go)

## Error Logs

> | Log Type | Node Type | Log Message | Description |

Reviewer: Can you sort contents by `Log Type`?

| ConsensusIstanbulCore | CN/PN/EN | **Drop an empty message from timeout channel** | when round change timer expires, this log is printed |
| NodeCN | CN | **fail to SendNewBlockHashes** err="write tcp 10.117.2.124:24108->10.117.2.108:32323: use of closed network connection" </br></br> **fail to SendNewBlockHashes** err="shutting down" | p2p peer disconnection |
| NetworksP2P | CN/PN/EN | **Protocol istanbul/64 failed** id=04680a827fa1b240 conn=staticdial err="write tcp 10.117.2.105:34396->10.117.2.27:32323: use of closed network connection"</br></br> **Protocol istanbul/64 failed** err="shutting down" | p2p peer disconnection |
| BlockchainTypes | EN | **decode anchor payload** err="rlp: expected input list for types.AnchoringDataLegacy" | Any type of value may be included in the data field of Anchoring tx. However, an error log is the output to the node when an incorrect type of value is entered. This log level seems too high, so we will consider the log level to lower down. |

## Warn Logs
| Log Type | Node Type | Log Message | Description |
| :--- | :--- | :--- | :--- |

> | Blockchain | CN/PN/EN | **Upgrade database version** from=N/A to=3 | It is logged at the beginning of the node start-up |

Reviewer: I think the description should provide either the more detailed information about why it happened or a guide for the further reaction. In my consideration, this description doesn't provide any valuable information.

# Node Log <a id="node-log"></a>

> This page details some important or frequently asked logs of the Klaytn node logs.

Reviewer 2: ```suggestion
This page details some important or frequently asked logs from Klaytn nodes.
```

If the Klaytn log is modified or newly added/deleted, please edit this page as well.

For more detailed information about log types, you can refer to [log_modules.go](https://github.com/klaytn/klaytn/blob/dev/log/log_modules.go)

> If you encounter any abnormal situation, please report it to the klaytn team: [klaytn issues](https://github.com/klaytn/klaytn/issues)

Reviewer 2: ```suggestion
If you encounter any abnormal situation, please report it to the klaytn team via [github](https://github.com/klaytn/klaytn/issues), [Klaytn Forum](https://forum.klaytn.foundation/), or [Discord](https://discord.io/KlaytnOfficial).
```

## Error Logs
| Log Type | Node Type | Log Message | Description | Suggested Guide |
| :--- | :--- | :--- | :--- | :--- |

> | Blockchain | CN/PN/EN |########## BAD BLOCK #########</br>Chain config: %v</br></br>Number: %v</br>Hash: 0x%x</br>%v</br></br>Error: %v</br>############################## | A bad block occurs when the received receipt and execution result is not matched. If a node stops with bad block log, there’s two cases. </br> - case 1. the configuration of the node is wrong such as binary version. </br> - case 2. there’s a problem with code. It is very likely that other nodes will also experience the same problem. | This error is critical, so if you see any bad block, please make an issue or report it to the klaytn github repository. |

Reviewer 3: ```suggestion
| Blockchain | CN/PN/EN |########## BAD BLOCK #########</br>Chain config: %v</br></br>Number: %v</br>Hash: 0x%x</br>%v</br></br>Error: %v</br>############################## | A bad block occurs when the received receipt and the execution result do not match. If a node stops with bad block log, it could be due to two reasons. </br> - Case 1. The configuration of the node is wrong such, as the binary version. </br> - Case 2. There’s a problem with the code. It is very likely that other nodes will also experience the same problem. | This error is critical, so if you see any bad block, please make an issue or report it to the Klaytn GitHub repository. |
```

> | ConsensusIstanbulCore | CN/PN/EN | **Drop an empty message from timeout channel** | It means round change timer expires. If the timer closes accidentally, this error is printed. | The error may occur when the downloader is started. </br> check next log is also printed: `Block synchronisation started`. |

Reviewer 3: ```suggestion
| ConsensusIstanbulCore | CN/PN/EN | **Drop an empty message from timeout channel** | It means that the round change timer will expire. This error is printed if the timer closes accidentally. | The error may occur when the downloader is started. </br> check next log is also printed: `Block synchronisation started`. |
```

> | NetworksP2P | CN/PN/EN | **Protocol istanbul/64 failed** id=04680a827fa1b240 conn=staticdial err="write tcp 10.117.2.105:34396->10.117.2.27:32323: use of closed etwork connection"</br></br> **Protocol istanbul/64 failed** err="shutting down" | This log can be printed when the other node is disconnected. Usually `Disconnected a P2P Peer` log is followed. | Check the disconnected peer is reconnected again. if not, check the network status or peer connection [admin_peers](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_peers) |

Reviewer 3: ```suggestion
| NetworksP2P | CN/PN/EN | **Protocol istanbul/64 failed** id=04680a827fa1b240 conn=staticdial err="write tcp 10.117.2.105:34396->10.117.2.27:32323: use of closed etwork connection"</br></br> **Protocol istanbul/64 failed** err="shutting down" | This log can be printed when the other node is disconnected. It is usually followed by`Disconnected a P2P Peer` log. | Check if the disconnected peer is reconnected again. If not, check the network status or peer connection [admin_peers](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_peers) |
```

Reviewer 3: @yoomee1313  여기서 if not은 disconnected peer가 reconnect 안 됐을 때를 말 하는 건가용??

Author: @terrikwak 넵 reconnect 안됬을때를 말합니다.

Reviewer 3: 이것도 수정했습니다

| NodeCN | CN | **fail to SendNewBlockHashes** err="write tcp 10.117.2.124:24108->10.117.2.108:32323: use of closed network connection" </br></br> **fail to SendNewBlockHashes** err="shutting down" | same as `Protocol istanbul/64 failed` | same as `Protocol istanbul/64 failed` |
| NodeCN | CN | **fail to SendNewBlock** peer=d35220eccdb0de7b err="shutting down" | same as `Protocol istanbul/64 failed` | same as `Protocol istanbul/64 failed` | 

> | NetworksRPC | EN (mostly) | **FastWebsocketHandler fail to upgrade message** error="websocket: version != 13" | Version issue of WebSocket connection | The header of the request should contain `Sec-Websocket-Version` field, and its value must be 13. You may not use klaytn rpc client. |

Reviewer 3: ```suggestion
| NetworksRPC | EN (mostly) | **FastWebsocketHandler fail to upgrade message** error="websocket: version != 13" | Version issue of WebSocket connection | The header of the request should contain `Sec-Websocket-Version` field with the value set at 13. Do not use klaytn rpc client. |
```

Author: @terrikwak 앗 여기서 You may not use klaytn rpc client 는, "너가 klaytn rpc client를 안썼을수도 있다." 를 표현한 것이었습니다.

Reviewer 3: 앗 네네!

## Warn Logs
| Log Type | Node Type | Log Message | Description | Suggested Guide |
| :--- | :--- | :--- | :--- | :--- |
| Blockchain | CN/PN/EN | **Upgrade database version** from=N/A to=3 | It is logged at the beginning of the node start-up | You don't need to handle this. |

> | ConsensusIstanbulCore | CN | **[RC]** round= | Round change log is started with [RC] tag. | If the round does not end in one or two rounds and continues to rise, then the network status or peer connection should be analyzed first. </br></br> peer connection check api: [admin_peers](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_peers)|

Reviewer 3: ```suggestion
| ConsensusIstanbulCore | CN | **[RC]** round= | Round change log is started with [RC] tag. | If the round does not end in one or two rounds and continues to go up, then the network status or peer connection should be analyzed first. </br></br> peer connection check api: [admin_peers](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_peers)|
```

| ConsensusIstanbulCore | CN | **unexpected request**  address= state="Accept request" seq=312 err="old message" number=311 hash=d960ea…6df6de | A proposer mines a block, but it is turned out unexpected. In most cases, it is too old to be a new block. | You don't need to handle this. |

> | Node | CN/PN/EN | **Failed doConnTypeHandshake** addr=10.117.2.252:28516 conn=inbound conntype=-1 err="read tcp 10.117.2.78:32324->10.117.2.252:28516: i/o timeout | By dialing, the two P2P peers setup a connection. If the setup fails, this log is printed. | Check the disconnected peer is reconnected again. if not, check the network status or peer connection </br></br> peer connection check api: [admin_peers](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_peers) |

Reviewer 3: ```suggestion
| Node | CN/PN/EN | **Failed doConnTypeHandshake** addr=10.117.2.252:28516 conn=inbound conntype=-1 err="read tcp 10.117.2.78:32324->10.117.2.252:28516: i/o timeout | By dialing, the two P2P peers setup a connection. This log is printed if the setup fails. | Check if the disconnected peer is reconnected again. If not, check the network status or peer connection </br></br> peer connection check api: [admin_peers](https://docs.klaytn.foundation/dapp/json-rpc/api-references/admin#admin_peers) |
```

| NodeCN | PN/EN | **Failed to filter bodies** peer=c02e4b4d471c56b9 lenTxs=1 | A node received the unwanted block header of body when fetching. </br> - lenTxs: non-requested number of txs | You don't need to handle this. |

> | Work | CN | **Transaction aborted due to time limit** hash= | The block execution time when mining should not exceed 250ms, so the last tx can be aborted due to this time limit. | Check the transaction enters the block eventually. |

Reviewer 3: ```suggestion
| Work | CN | **Transaction aborted due to time limit** hash= | The block execution time when mining should not exceed 250ms, so the last transaction can be aborted due to this time limit. | Confirm that the transaction enters the block. |
```

> | Work | CN | **Transaction failed, account skipped** hash=b1b26c...6b220a err="insufficient balance for transfer"</br></br>Error(before v1.6.2)</br>Warn(after v1.6.2) | When a transaction cannot be executed when mining due to an insufficient balance of the "from account" </br> (Theoretically, it occurs when the balance was sufficient at the time when the transaction was created and entered the txpool, but there was no balance at the actual execution time.) | Check `from` account is really out of balance. |

Reviewer 3: ```suggestion
| Work | CN | **Transaction failed, account skipped** hash=b1b26c...6b220a err="insufficient balance for transfer"</br></br>Error(before v1.6.2)</br>Warn(after v1.6.2) | When a transaction cannot be executed during mining due to an insufficient balance in the `from` account </br> (Theoretically, it occurs when the balance was sufficient at the time when the transaction was created and entered the txpool, but not at the actual execution time.) | Check if the `from` account is really out of balance. |
```

## Info Logs

> `Info` log contains the additional information so that you can know more about the node status, so you don't need to handle `Info` level log.

Reviewer 3: ```suggestion
`Info` log contains the additional information for you to know more about the node status, so you don't need to handle `Info` level log.
```

| Log Type | Node Type | Log Message | Description |
| :--- | :--- | :--- | :--- |

> | Blockchain | CN/PN/EN | **Regenerated local transaction journal** transactions=0 accounts=0 | When node is shutdown, local txs are journaled to a file (default file name is transactions.rlp). When node is restarted with the journaled file, the local txs can be regenerated based on the file. </br> - transactions: number of the regenerated local transaction. </br> - accounts: number of the regenerated accounts(==from) |

Reviewer 3: ```suggestion
| Blockchain | CN/PN/EN | **Regenerated local transaction journal** transactions=0 accounts=0 | When node is shut down, local txs are journaled to a file (default file name is transactions.rlp). When node is restarted with the journaled file, the local transactions can be regenerated based on the file. </br> - transactions: number of the regenerated local transaction. </br> - accounts: number of the regenerated accounts(==from) |
```

> | Blockchain | CN/PN/EN| **Inserted a new block** number=14 hash=13cbfc…f007fc txs=0 gas=0 elapsed=793.458µs  processTxs=167ns finalize=157.708µs validateState=7.542µs totalWrite=443.417µs trieWrite=256.667µs | If the node is not a proposer at that block, and consensus is succeed, the node have executed(==validates) the block. In other words, a block is inserted. </br> - gas: total gas spent during tx execution. </br> This field is commonly used when testing the network to figure out the used gas per block. |

Reviewer 3: ```suggestion
| Blockchain | CN/PN/EN| **Inserted a new block** number=14 hash=13cbfc…f007fc txs=0 gas=0 elapsed=793.458µs  processTxs=167ns finalize=157.708µs validateState=7.542µs totalWrite=443.417µs trieWrite=256.667µs | If the node is not a proposer at that block, and the consensus succeeds, the node have executed(==validates) the block. In other words, a block is inserted. </br> - gas: total gas spent during tx execution. </br> This field is commonly used when testing the network to find the used gas per block. |
```

> | NetworksP2P | CN/PN/EN | **[Dial] Add dial candidate from static nodes**  id=62a08a4b9f091c4b NodeType=0 ip=10.117.2.8 mainPort=32323 port=[32323] | A new P2P peer is connected, and it is a static node. A node added by manually by using static-nodes.json or addpeer api is called static node. If multichannel, it uses two port. ex. [32323, 32324]. </br> - id: dst peer id </br> - NodeType: dst node type(cn,pn,en,bn) </br> - ip: dst ip </br> - mainPort: dst TCP listening port number </br> - port: dst TCP listening port number including both main port and subports.|

Reviewer 3: ```suggestion
| NetworksP2P | CN/PN/EN | **[Dial] Add dial candidate from static nodes**  id=62a08a4b9f091c4b NodeType=0 ip=10.117.2.8 mainPort=32323 port=[32323] | A new P2P peer is connected, and it is a static node. A node added manually by using static-nodes.json or addpeer api is called static node. If it is a multichannel, it uses two ports. ex. [32323, 32324]. </br> - id: dst peer id </br> - NodeType: dst node type(cn,pn,en,bn) </br> - ip: dst ip </br> - mainPort: dst TCP listening port number </br> - port: dst TCP listening port number including both the main port and sub port.|
```

> | NetworksP2P | CN/PN/EN | **Added a multichannel P2P Peer** id=28a6760472a078fb conn=staticdial peerID=28a6760472a078fb | A new peer is connected as a multichannel peer. </br> - id/peerID: my node’s peer id </br> - conn: type of connection </br> - inbound: somebody connects me </br> - staticdial: static connection, such as static-nodes.json or addPeer </br> - trusteddial: trusted connection, such as trust-nodes.json. it is always reconnected and  the trusted connections should not be count toward the maximum number of connections.|

Reviewer 3: ```suggestion
| NetworksP2P | CN/PN/EN | **Added a multichannel P2P Peer** id=28a6760472a078fb conn=staticdial peerID=28a6760472a078fb | A new peer is connected as a multichannel peer. </br> - id/peerID: my node’s peer id </br> - conn: type of connection </br> - inbound: somebody connects me </br> - staticdial: static connection, such as static-nodes.json or addPeer </br> - trusteddial: trusted connection, such as trust-nodes.json. It is always reconnected and the trusted connections should not be counted toward the maximum number of connections.|
```

Author: @terrikwak 넵넵 그러니까, "커넥션 max 개수를 넘었는지 계산할 때 trusted connection가 포함되면 안된다"라는 의미입니다. 아마 다른곳에 쓰여진 걸 그대로 들고 와서 그런것 같은데.. 음, 고쳐놓겠습니다!

Author: 고쳤습니다. https://github.com/klaytn/klaytn-docs/pull/443/commits/afd912ba15f62abc4e22401119496f679d2d1017

> | NetworksP2P | CN/PN/EN | **Disconnected a multichannel P2P Peer** id=28a6760472a078fb conn=inbound    peerID=28a6760472a078fb peerName=Klaytn/v1.7.3+acae89350c/darwin-arm64/go1.18.1 err=EOF | A multichannel peer is disconnected. </br> - peerName: it shows my node info </br> - err: the reason why the connection is disconnected |

Reviewer 3: ```suggestion
| NetworksP2P | CN/PN/EN | **Disconnected a multichannel P2P Peer** id=28a6760472a078fb conn=inbound    peerID=28a6760472a078fb peerName=Klaytn/v1.7.3+acae89350c/darwin-arm64/go1.18.1 err=EOF | A multichannel peer is disconnected. </br> - peerName: It shows my node info </br> - err: The reason why the connection is disconnected |
```

| NetworksP2P | CN/PN/EN | **ProtocolManager.processConsensusMsg closed** id=28a6760472a078fb conn=inbound    PeerName=Klaytn/v1.7.3+acae89350c/darwin-arm64/go1.18.1 | When a P2P node is disconnected, the consensus message channel between them is closed, too. |

> | StorageStateDB | CN/PN/EN | **Persisted trie from memory database** blockNum=23460 updated nodes=4 updated nodes size=499.00B time=539.959µs  gcnodes=68  gcsize=10.55kB gctime=226.499µs  livenodes=245 livesize=37.80kB | This log is printed to inform you that trie db has been committed. In here, commit means flushing db change to the actual db. </br></br> commit is done periodically. </br> - case1. if the node is full node, trie commit is done in every 128 block. </br> - case2. if the node is an archive node, trie commit is done in every single block. </br></br> commit is done at next circumstances, too. </br> - when node is shutdown, commit is done. </br> - when memory size exceeds the cap, commit is done. </br></br> Tip. </br> - gc stands for garbage collection. Here, garbage collection means flushing db writes of trie node change. </br> - full node stores the information of every 128 cycles and the latest 128 blocks. </br> - archive node stores the information of an every block.|

Reviewer 3: ```suggestion
| StorageStateDB | CN/PN/EN | **Persisted trie from memory database** blockNum=23460 updated nodes=4 updated nodes size=499.00B time=539.959µs  gcnodes=68  gcsize=10.55kB gctime=226.499µs  livenodes=245 livesize=37.80kB | This log is printed to inform you that trie db has been committed. Here, commit means flushing db change to the actual db. </br></br> Commit is done periodically. </br> - Case 1. If the node is a full node, trie commit is done for every 128 block. </br> - Case 2. If the node is an archive node, trie commit is done for every block. </br></br> Commit is done at next circumstances, too. </br> - When a node is shut down, commit is done. </br> - When memory size exceeds the cap, commit is done. </br></br> Tip. </br> - gc stands for garbage collection. Here, garbage collection means flushing db writes of trie node change. </br> - Full node stores the information of every 128 cycles and the latest 128 blocks. </br> - Archive node stores the information of every block.|
```

Reviewer 3: @yoomee1313  at next circumstances 는 무슨 뜻인가요?

Author: @terrikwak "다음상황에서" 라는 의미로 썼고, 뒤에 나오는 줄들이 그 상황을 설명하고 있습니다.
